File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        d1 = {}
        d2 = {}

        if len(s) != len(t):
            return False

        for x in s:
            d1[x] = d1.get(x, 0) + 1
        
        for x in t:
            d2[x] = d2.get(x, 0) + 1
        
        if d1 == d2:
            return True
        else:
            return False

File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    maxArea(heights) {
        let start = 0, end = heights.length - 1;
        let maxArea = 0;

        while (start < end) {
            let area = (end - start) * (Math.min(heights[end], heights[start]));

            if (maxArea < area) {
                maxArea = area;
            }

            if (heights[end] > heights[start]) {
                start++;
            } else {
                end--;
            }

        }

        return maxArea;
    }
}
